### PR TITLE
Some Kotlin refactoring while writing the BDD

### PR DESCRIPTION
**Pull request by nickmcmahon01**
Rename a method and take the shorter function style

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstier/service/ChangeLevelCalculator.kt [hunk 1]
```diff
@@ -12,48 +12,40 @@ class ChangeLevelCalculator(
...
-    }
-    return TierLevel(ChangeLevel.ZERO, 0)
-  }
-
-  private fun getIomNominal(crn: String): Int {
-    // We don't care about the full list, only if there is IOM Nominal
-    return if (communityApiDataService.getComplexityFactors(crn).any { it == ComplexityFactor.IOM_NOMINAL }) 2 else 0
+    return if (hasMandateForChange(crn)) {
+      val totalPoints = getOasysNeedsPoints(crn) + getOgrsPoints(crn) + getIomNominalPoints(crn)
+      val tier = when {
```
Comment (nickmcmahon01) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstier/service/ChangeLevelCalculator.kt:17]: now all the places we decide the tier level are in one place.
we'll also be adding an extra on here for the out of date assesment check

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstier/service/ChangeLevelCalculator.kt [hunk 2]
```diff
@@ -12,48 +12,40 @@ class ChangeLevelCalculator(
...
+      TierLevel(tier, totalPoints)
+    } else TierLevel(ChangeLevel.ZERO, 0)
   }
 
-  private fun shouldCalculateChangeLevel(crn: String): Boolean =
+  private fun hasMandateForChange(crn: String): Boolean =
     when {
-      communityApiDataService.isCurrentCustodialSentence(crn) -> true
-      communityApiDataService.isCurrentNonCustodialSentence(crn) -> hasNoUnpaidWorkOrRestrictiveRequirements(crn)
+      communityApiDataService.hasCurrentCustodialSentence(crn) -> true
```
Comment (nickmcmahon01) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstier/service/ChangeLevelCalculator.kt:28]: Happy to change back, but I like to use has if we're checking a collection. i.e 'has any current custodial sentence'

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstier/service/ChangeLevelCalculator.kt [hunk 3]
```diff
@@ -12,48 +12,40 @@ class ChangeLevelCalculator(
...
+    assessmentApiDataService.getAssessmentNeeds(crn).let {
       it.entries.sumBy { ent ->
         ent.key.weighting.times(ent.value?.score ?: 0)
       }
     }
-  }
 
-  private fun getOgrsPoints(crn: String): Int {
-    return communityApiDataService.getOGRS(crn).let {
+  private fun getOgrsPoints(crn: String): Int =
```
Comment (nickmcmahon01) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstier/service/ChangeLevelCalculator.kt:43]: I've come round to liking this style for simple clauses like this

**Review by markreesmoj — APPROVED** [review]
👍

**Pull request merged**